client/daemon/peer/piece_manager.go

```diff
@@ -965,3 +1065,88 @@ func (pm *pieceManager) downloadPieceFromSource(ctx context.Context,
 	pt.PublishPieceInfo(pieceNum, uint32(result.Size))
 	return nil
 }
+
+func (pm *pieceManager) downloadPieceGroupFromSource(ctx context.Context,
+	pt Task, log *logger.SugaredLoggerOnWith,
+	peerTaskRequest *schedulerv1.PeerTaskRequest,
+	pieceSize uint32, pg *pieceGroup,
```

Comment:
and maybe also encapsulate the param pieceGroup related field like pieceSize, parsedRange into pieceGroup make this method cleaner